### PR TITLE
Automated refactors from baseline 1.10.1 (does not upgrade baseline)

### PR DESCRIPTION
**Pull request by carterkozak**
**Goals (and why)**:
Primarily for safe logging data, also to fix bugs that the static analysis tooling knows how to resolve.

Methodology:
1. Set baseline version to 1.10.1 (not committed)
1. `./gradlew compileJava compileTestJava -PrefasterApply=true --no-build-cache`
1. `./gradlew compileJava compileTestJava -PerrorProneApply=true --no-build-cache`
1. Run the idea optimize imports task on modified files
1. run the idea 'Unnecessary fully qualified name' inspection, and apply fixes to resolve logsafe preconditions fqcns.

### CODE REVIEW

File: atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Cell.java [hunk 1]
```diff
@@ -59,8 +59,8 @@ public static boolean isNameValid(byte[] name) {
     }
 
     private void validateNameValid(byte[] name) {
-        Preconditions.checkNotNull(name, "name cannot be null");
-        Preconditions.checkArgument(name.length > 0, "name must be non-empty");
+        com.palantir.logsafe.Preconditions.checkNotNull(name, "name cannot be null");
+        com.palantir.logsafe.Preconditions.checkArgument(name.length > 0, "name must be non-empty");
```
Comment (felixdesouza) [inline atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Cell.java]: why do we still require both `Preconditions` in this file? Unclear to me why the invocation on line 66 can't be converted also?

**Comment by carterkozak** [comment]
The idea is the migration to safe logging preconditions does as much as possible without risk -- we want to be careful not to make code less correct.

In the example `(guava)Preconditions.checkNotNull(value, "Hello, %s!", name)`, the output may be worse if we refactor to `(safe)Preconditions.checkNotNull(value, "Hello, %s!", UnsafeArg.of("name", name))` because safe-logging preconditions have no concept of interpolation.

File: atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Namespace.java [hunk 2]
```diff
@@ -58,7 +57,7 @@ public static Namespace create(String name) {
     @SuppressWarnings("ValidateConstantMessage") // https://github.com/palantir/gradle-baseline/pull/175
     public static Namespace create(String name, Pattern pattern) {
         Validate.notEmpty(name, "namespace name cannot be empty (see Namespace.EMPTY_NAMESPACE instead).");
-        Validate.isTrue(!name.contains("."), "namespace cannot contain dots (atlas reserved).");
+        Preconditions.checkArgument(!name.contains("."), "namespace cannot contain dots (atlas reserved).");
```
[the remark this answers is not in the document]
Comment (carterkozak) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Namespace.java]: The next line is parameterized so we cannot safely automate migration to safe-logging.

[the remark this answers is not in the document]
Comment (felixdesouza) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Namespace.java]: yeah, I realised what was going on in the end :)

**Comment by carterkozak** [comment]
> I'm not sure what value we get by just applying safe preconditions to constant messages.

It allows us to debug more quickly without having to reference code by line number. By moving as much existing use to safe-logging, we push developers to use it in a first class way for new code. I think having some inconsistent files that use both guava and safe-logging preconditions is a reasonable price for the debugging data that becomes available.

**Comment by felixdesouza** [comment]
> It allows us to debug more quickly without having to reference code by line number.

I'm not sure what this means, I haven't looked at the impl of safe logging in a while. What difference is there between constant message that is with preconditions vs safe logged? is it that the message is actually visible?

Note, I'm not averse to merging this in, I'm just curious about a few things, so thanks for this!

**Comment by carterkozak** [comment]
> What difference is there between constant message that is with preconditions vs safe logged? is it that the message is actually visible?

You are correct, with safe-logging preconditions we get:
```
com.palantir.logsafe.exceptions.SafeNullPointerException: constant message
	at com.palantir.logsafe.Preconditions.checkNotNull(Preconditions.java:207)
	at com.palantir.Main.main(Main.java:14)
```

Where guava throws a NPE that is not marked safe:
```
java.lang.NullPointerException: {throwable0_message}
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:895)
	at com.palantir.Main.main(Main.java:14)
```
with an unsafe parameter that may or may not be accessible depending on your environment:
```
{throwable0_message: constant message}
```

The constnat-string check for commons-lang3 validate and guava preconditions exists for performance (we want to avoid expensive string concatenation/formatting on the fast path, and only execute it when validation fails). There's unfortunately no way  to leverage it in safe-logging.

**Comment by felixdesouza** [comment]
I think some other feedback for the logsafe refaster plugin, is probably switching the imports i.e. fully qualify the guava one as opposed to the logsafe one. Rationale being, if the guava one remains, then you will likely have it split over multiple lines as you're doing interpolation, so fully qualifying `com.google.common.base.Preconditions.checkState` wouldn't cause checkstyle to freak out.

**Pull request merged**